tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -1152,6 +1152,40 @@ public function testModelAttributesAreCastedWhenPresentInCastsArray()
 	}
 
 
+	public function testModelAttributeCastingPreservesNull()
```

Comment:
Grrr. I really hate laravel's tests like this. Way to much in a single test. We should be using data providers or something.